application/src/main/java/org/togetherjava/tjbot/commands/help/HelpThreadActivityUpdater.java

```diff
@@ -78,7 +89,20 @@ private void updateActivityForThread(ThreadChannel threadChannel) {
 
     private static RestAction<HelpSystemHelper.ThreadActivity> determineActivity(
             MessageChannel channel) {
-        return channel.getHistory().retrievePast(ACTIVITY_DETERMINE_MESSAGE_LIMIT).map(messages -> {
+        Long mostRecentMessageId =
+                manuallyResetChannelActivityCache.getIfPresent(channel.getIdLong());
+        RestAction<List<Message>> restActionMessages;
+
+        if (mostRecentMessageId != null) {
+            MessageHistory.MessageRetrieveAction historyAfter =
+                    channel.getHistoryAfter(mostRecentMessageId, ACTIVITY_DETERMINE_MESSAGE_LIMIT);
+            restActionMessages = historyAfter.map(MessageHistory::getRetrievedHistory);
+        } else {
+            restActionMessages =
+                    channel.getHistory().retrievePast(ACTIVITY_DETERMINE_MESSAGE_LIMIT);
+        }
```

Comment:
id move this into a helper method, such as `getRelevantHistory`

Reply:
done.